packages/sns/lib/utils/snsInitter.ts

```diff
@@ -97,9 +132,10 @@ export async function deleteSnsSqs(
   snsClient: SNSClient,
   deletionConfig: DeletionConfig,
   queueConfiguration: CreateQueueCommandInput,
-  topicConfiguration: CreateTopicCommandInput,
+  topicConfiguration: CreateTopicCommandInput | undefined,
   subscriptionConfiguration: SNSSubscriptionOptions,
   extraParams?: ExtraParams,
+  topicLocator?: SNSTopicLocatorType,
```

Comment:
Couldn't we rework this method and introduce a breaking change?

Reply:
potentially. if you see a clear way to do this that would simplify the code, go for it